src/main/kotlin/org/kamiblue/client/util/CircularArray.kt

```diff
@@ -0,0 +1,56 @@
+package org.kamiblue.client.util
+
+class CircularArray<T : Number> constructor(private val array: Array<T>, filled: Boolean) : Iterable<T> {
```

Comment:
Shouldn't this be implemented on the MutableList interface? Since it has random access by index.

Reply:
We can't use the MutableList interface, once I make the iterator an inner class then the get interface will be gone since we want to allow a "push" and a forward only access due, I was going to make it less generic, but I figured I would stick with the RingBuffer FIFO, but allowing overwrites which I can't find any interface that would fit with that, Also, FloatArray was the original implementation and that doesn't have an interface I can find in Kotlin at all so I could just implement Array in a sense.

Reply:
> We can't use the MutableList interface, once I make the iterator an inner class then the get interface will be gone since we want to allow a "push" and a forward only access due, I was going to make it less generic, but I figured I would stick with the RingBuffer FIFO, but allowing overwrites which I can't find any interface that would fit with that, Also, FloatArray was the original implementation and that doesn't have an interface I can find in Kotlin at all so I could just implement Array in a sense.

Well then what about MutableCollection?

Reply:
Is there a reason for defining the interface? FloatArray doesn't and this isn't in a shared library. I will do it if it would add anything. I'm avoiding the collections interface and reimplementing average for performance gains around keeping the data in cache. The average generic version requires copying the data into a list unless it's specialized on a specific array interface I believe. I would need to double check it when I'm not out and about.

Reply:
I don't see a valuable reason for defining it ourselves if it's not in the commons repo, ie shared library.